lib/semian/mysql2.rb

```diff
@@ -0,0 +1,57 @@
+class Mysql2::SemianError < Mysql2::Error
+  def initialize(semian_identifier, *args)
+    super(*args)
+    @semian_identifier = semian_identifier
+  end
+
+  def to_s
+    "[#{@semian_identifier}] #{super}"
+  end
+end
+
+module Semian
+  module Mysql2
+    class << self
+      attr_accessor :default_options
+
+      def enable!(default_options = {})
+        self.default_options = default_options
+        ::Mysql2::Client.prepend(self)
+      end
+    end
+
+    def semian_identifier
+      @semian_identifier ||= begin
+        name = query_options[:semian_resource]
+        name ||= [query_options[:host] || 'localhost', query_options[:port] || 3306].join(':')
+        :"mysql_#{name}"
+      end
+    end
+
+    def query(*)
+      semian_resource.acquire { super }
+    rescue ::Semian::BaseError => error
+      raise ::Mysql2::SemianError.new(semian_identifier, error)
+    end
+
+    private
+
+    def connect(*)
+      semian_resource.acquire { super }
+    rescue Semian::BaseError => error
+      raise ::Mysql2::SemianError.new(semian_identifier, error)
+    end
+
+    def semian_resource
+      @semian_resource ||= ::Semian.register(semian_identifier, **semian_options)
+    end
+
+    def semian_options
+      options = ::Semian::Mysql2.default_options.dup
```

Comment:
Hum, if you don't set the global options, and don't set the `:semian` in the `database.yml` either, you'll end up with `0` tickets. Meaning it will raise.

I start to wonder if the global options actually make sense. I think I'll get rid of it, and warn is no options are present.